### DISCUSSION TITLE
How to maintain good code quality and data quality of a Data Pipeline?

### DISCUSSION THREAD

**Discussion by electro-drag** [community/community#134660] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

General

### Body

I have a data pipeline which has python in backend. I wanted to explore on the ways that how can I have a good code quality as well as data quality in the project.

**Comment by shloktech** [comment]
I would suggest the following:
For Code Quality: Sonar Qube (https://www.sonarsource.com/products/sonarqube/)
For Security: Mend (https://www.mend.io/)
For Data Testing: DBT (https://docs.getdbt.com/docs/build/data-tests)

**Reply by electro-drag** [reply]
I have never heard of DBT what is it?

**Reply by shloktech** [reply]
It's a very powerful tool :)
DBT stand for Data Build Tool and it is used for transforming and testing your data efficiently.
Please refer to below resources: 
What is dbt? : https://www.getdbt.com/product/what-is-dbt
Data testing with dbt: https://www.getdbt.com/analytics-engineering/transformation/data-testing

**Reply by electro-drag** [reply]
Nice I got to learn about a new technology today :) I would be looking forward on exploring more on it and using it in my project. Thanks a lot @shloktech
